core/src/main/java/com/linecorp/armeria/internal/common/DefaultSplitHttpRequest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.common;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.SplitHttpRequest;
+
+import io.netty.util.concurrent.EventExecutor;
+
+public final class DefaultSplitHttpRequest extends AbstractSplitHttpMessage implements SplitHttpRequest {
+
+    private final RequestHeaders headers;
+
+    public DefaultSplitHttpRequest(HttpRequest request, EventExecutor executor) {
+        super(request, executor, new BodySubscriber(0, request, executor));
+        headers = request.headers();
+    }
+
+    @Override
+    public RequestHeaders headers() {
+        return headers;
+    }
```

Comment:
Could you add `toString()`?

```java
return MoreObjects.toStringHelper(this)
                  .add("headers", headers)
                  .add("upstream", request)
                  .toString();
```

Reply:
Okay :-) I added in https://github.com/line/armeria/pull/3953/commits/b5c711da0b228cd758e1c50cb0a679e08dfd61af